src/gc/gc.d

```diff
@@ -1887,6 +1894,59 @@ struct Gcx
         return pool.npages * PAGESIZE;
     }
 
+    /**
+     * Update the thresholds for when to collect the next time
+     */
+    void updateCollectThresholds() nothrow
+    {
+        static float max(float a, float b) nothrow
+        {
+            return a >= b ? a : b;
+        }
+
+        static size_t roundUp(float val, size_t multiple) nothrow
+        {
+            return cast(size_t)(val + multiple - 1 + 0.5) / multiple * multiple;
+        }
+
+        static float zeroDenormal(float val) nothrow
+        {
+            return val + float.min_normal - float.min_normal;
```

Comment:
Ah, the DC offset should be `float.min_normal / float.epsilon`, so that the add and subtract zero out the number.
It can't be optimized away because of the side effects and it's necessary because you might free all large memory at some point.